be/src/exec/hdfs_scanner_text.cpp

```diff
@@ -253,80 +249,70 @@ Status HdfsTextScanner::parse_csv(int chunk_size, ChunkPtr* chunk) {
+
+            size_t chunk_index = _scanner_params.materialize_index_in_chunk[j];
+            size_t csv_index = _formatted_hive_column_name_2_index[formatted_slot_name];
+            Column* column = _column_raw_ptrs[chunk_index];
+            if (csv_index < fields.size()) {
+                const Slice& field = fields[csv_index];
                 options.type_desc = &(_scanner_params.materialize_slots[j]->type());
                 if (!_converters[j]->read_string(column, field, options)) {
-                    LOG(WARNING) << "Converter encountered an error for field " << field.to_string() << ", index "
-                                 << index << ", column " << _scanner_params.materialize_slots[j]->debug_string();
-                    error_msg = strings::Substitute("CSV parse column [$0] failed, more details please see be log.",
-                                                    _scanner_params.materialize_slots[j]->debug_string());
-                    chunk->get()->set_num_rows(num_rows);
-                    has_error = true;
-                    break;
+                    return Status::InternalError(
+                            strings::Substitute("CSV converter encountered an error for field: $0, column name is: $1",
+                                                field.to_string(), formatted_slot_name));
                 }
             } else {
                 // The size of hive_column_names may be larger than fields when new columns are added.
                 // The default value should be filled when querying the extra columns that
                 // do not exist in the text file.
-                // hive only support null column
-                // TODO: support not null
-                column->append_nulls(1);
+                column->append_default();
             }
         }
-        num_rows += !has_error;
-        if (!has_error) {
-            // Partition column not stored in text file, we should append these columns
-            // when we select partition column.
-            int num_part_columns = _scanner_ctx.partition_columns.size();
-            for (int p = 0; p < num_part_columns; ++p) {
-                int index = _scanner_params.partition_index_in_chunk[p];
-                Column* column = _column_raw_ptrs[index];
-                ColumnPtr partition_value = _scanner_ctx.partition_values[p];
-                DCHECK(partition_value->is_constant());
-                auto* const_column = ColumnHelper::as_raw_column<ConstColumn>(partition_value);
-                const ColumnPtr& data_column = const_column->data_column();
-                if (data_column->is_nullable()) {
-                    column->append_nulls(1);
-                } else {
-                    column->append(*data_column, 0, 1);
-                }
+
+        // Start to append partition column
+        for (size_t p = 0; p < _scanner_ctx.partition_columns.size(); ++p) {
```

Comment:
I remember in hdfs scanner context, there is a method to add partitioned columns.

Reply:
But `HdfsScannerContext::append_partition_column_to_chunk()` can't control the partition column's position in a chunk.

Reply:
in hive_connector.cpp, we reorder columns. 

```cpp

Status HiveDataSource::get_next(RuntimeState* state, ChunkPtr* chunk) {
    if (_no_data) {
        return Status::EndOfFile("no data");
    }
    _init_chunk(chunk, _runtime_state->chunk_size());
    do {
        RETURN_IF_ERROR(_scanner->get_next(state, chunk));
    } while ((*chunk)->num_rows() == 0);

    // The column order of chunk is required to be invariable. In order to simplify the logic of each scanner,
    // we force to reorder the columns of chunk, so scanner doesn't have to care about the column order anymore.
    // The overhead of reorder is negligible because we only swap columns.
    ChunkHelper::reorder_chunk(*_tuple_desc, chunk->get());

    return Status::OK();
}
```

Reply:
Perhaps you can move this function to `hdfs_scanner.cpp`.

Reply:
unify it in the future, otherwise, this pr is too large.